tests/Template/ClassTemplateTest.php

```diff
@@ -4207,6 +4207,68 @@ public function aggregate(...$values): null|int|float
                 'ignored_issues' => [],
                 'php_version' => '8.0',
             ],
+            'inheritedConditionalsWithMatchingTypes' => [
+                'code' => '<?php
+                    /** @template InstanceType */
+                    interface PluginManagerInterface
+                    {
+                        /**
+                         * @template TRequestedInstance extends InstanceType
```

Comment:
I don't think we support `extends` here

Reply:
it works flawless if I do not overwrite the method in the abstract class. the issue is adding the object return type which makes it less specific.

Reply:
Well, I confirm we don't do anything with the extends keyword so it's just comments as far as Psalm is concerned

Reply:
but it actually seems to work until I add the overridden method with the object return type 🤔

Reply:
can you show an example where psalm behaves differently based on the extends?

Reply:
I guess I cant. I just know that the way psalm behaves differs once I extend the `get` method and narrow the type from mixed to object

Reply:
could you post two snippets so we see the difference exactly? I'm still a bit confused

Reply:
Okay, I've digged a bit deeper and found out that there is a general issue with `@inheritdoc` as due to this, templates are completely dropped and therefore the whole type detection seems to be problematic.

https://psalm.dev/r/538a0d30d4

I am totally lost here, but thats maybe due to the fact that I spent too much time on other psalm related stuff in the past days.
Would really love to get some help on this but will also try to solve this on my own in the next weeks.